lefthk/src/config/keybind.rs

```diff
@@ -0,0 +1,107 @@
+pub struct Keybind {
+    pub command: Command,
+    pub modifier: Vec<String>,
+    pub key: Key,
+}
+
+impl TryFrom<Keybind> for Vec<core_keybind> {
+    type Error = LeftError;
+
+    fn try_from(kb: Keybind) -> Result<Self> {
+        let command_key_pairs: Vec<(Box<dyn core_command>, String)> = match kb.command {
+            Command::Chord(children) if !children.is_empty() => {
+                let key = get_key!(kb.key);
+                let children = children
+                    .iter()
+                    .filter_map(|kb| match TryFrom::try_from(kb.clone()) {
+                        Ok(keybinds) => Some::<Vec<lefthk_core::config::Keybind>>(keybinds),
+                        Err(err) => {
+                            tracing::error!("Invalid key binding: {}\n{:?}", err, kb);
+                            None
+                        }
+                    })
+                    .flatten()
+                    .collect();
+
+                vec![(Box::new(command_mod::Chord::new(children)), key)]
+            }
+            Command::Chord(_) => return Err(LeftError::ChildrenNotFound),
+            Command::Execute(value) if !value.is_empty() => {
+                let keys = get_key!(kb.key);
+                vec![(Box::new(command_mod::Execute::new(value)), keys)]
+            }
+            Command::Execute(_) => return Err(LeftError::ValueNotFound),
+            Command::Executes(values) if !values.is_empty() => {
+                let keys = get_keys!(kb.key);
+                if keys.len() != values.len() {
+                    return Err(LeftError::NumberOfKeysDiffersFromValues);
+                }
+                values
+                    .iter()
+                    .enumerate()
+                    .map(|(i, v)| {
+                        (
+                            Box::new(command_mod::Execute::new(v.to_owned()))
+                                as Box<dyn core_command>,
+                            keys[i].clone(),
+                        )
+                    })
+                    .collect()
+            }
+            Command::Executes(_) => return Err(LeftError::ValuesNotFound),
+            Command::ExitChord => {
+                let keys = get_key!(kb.key);
+                vec![(Box::new(command_mod::ExitChord::new()), keys)]
+            }
+            Command::Reload => {
+                let keys = get_key!(kb.key);
+                vec![(Box::new(command_mod::Reload::new()), keys)]
+            }
+            Command::Kill => {
+                let keys = get_key!(kb.key);
+                vec![(Box::new(command_mod::Kill::new()), keys)]
+            }
+        };
+        let keybinds = command_key_pairs
+            .iter()
+            .map(|(c, k)| core_keybind {
+                command: c.normalize(),
+                modifier: kb.modifier.clone(),
+                key: k.to_owned(),
+            })
+            .collect();
+        Ok(keybinds)
+    }
```

Comment:
I didn't change that much here, because the `Executes` command is a little bit complicated to change here. May I ask if I can fix that in another PR?

Reply:
(I suppose yes)